### PR TITLE
[GH-47] [FIX] Agenda item number should get updated if the items of the agenda are deleted

### LINKED ISSUE #47: Agenda item number should get updated if the items of the agenda are deleted
The agenda items' numbering gets off by one when the first item is deleted. 
The numbering is also off when the items are manually edited but might be difficult to tackle as Maria Nuñez has mentioned [here](https://github.com/mattermost/mattermost-plugin-agenda/pull/44#issuecomment-649738955)

QA Test Steps
1. Ensure that the [agenda plugin](https://github.com/mattermost/mattermost-plugin-agenda) is installed and enabled.
2. Go to a channel header and edit the agenda settings to update the hashtag if needed (e.g.: `test-Jan02`).
3. Use command `/agenda queue` to queue at least 3 items as following:
`/agenda queue item 1`
`/agenda queue item 2`
`/agenda queue item 3`
4. Click on the agenda hashtag to open up the agenda list on the RHS.
5. On the RHS, click the post menu for `item 1` (or the first item that was queued) and delete the agenda item.
6. Ensure that now there are only 2 items for that agenda hashtag: `item 2` and `item 3` on the RHS.
7. Add a new agenda item: `/agenda queue new item`.
8. Click on the hashtag to refresh the RHS.

Observed: The RHS has 3 agenda items numbered 2), 3) and 3)

Expected: The RHS has 3 agenda items numbered 1), 2) and 3)

Jira: https://mattermost.atlassian.net/browse/MM-26536

### PR DESCRIPTION
**Pull request by sanjaydemansol**
#### Summary
Fixes #47 
Now, Whenever we add new agenda item. Plugin searches for old messages for that hashtag. 
If some item is missing(deleted), it fixes numbering for all items. 

#### Testing
Ensure that the agenda plugin is installed and enabled.
Go to a channel header and edit the agenda settings to update the prefix and 
Use command /agenda queue to queue at least 3 items as following:
`/agenda queue item 1`
`/agenda queue item 2`
`/agenda queue item 3`

Click on the agenda hashtag to open up the agenda list on the RHS.
On the RHS, click the post menu for item 1 (or the first item that was queued) and delete the agenda item.
**Ensure that now there are only 2 items for that agenda hashtag: item 2 and item 3 on the RHS.**
Add a new agenda item: /agenda queue new item.
Click on the hashtag to refresh the RHS.

Outcome: The RHS has 3 agenda items numbered 1), 2) and 3)

#### Ticket Link
https://github.com/mattermost/mattermost-plugin-agenda/issues/47
****

### CODE REVIEW

File: server/command.go [hunk 1]
```diff
@@ -147,31 +164,147 @@ func (p *Plugin) executeCommandQueue(args *model.CommandArgs) *model.CommandResp
...
+	local, _ := time.LoadLocation("Local")
+	formattedDate, _ := time.ParseInLocation(hashtagDateFormat, originalPostDate, local)
+	if formattedDate.Year() == 0 {
+		thisYear := today.Year()
+		formattedDate = formattedDate.AddDate(thisYear, 0, 0)
+	}
 
-	_, appErr = p.API.CreatePost(&model.Post{
+	if today.Year() <= formattedDate.Year() && today.YearDay() < formattedDate.YearDay() {
+		return responsef("We don't support re-queuing future items, only available for present and past items.")
```
Comment (maisnamrajusingh) [inline server/command.go:256]: I think you should make it more short. Something like `Re-queing future items items are not supported.`

File: webapp/src/actions/index.js [hunk 2]
```diff
@@ -80,4 +80,34 @@ export function performSearch(terms) {
...
+    };
+}
+
+export async function clientExecuteCommand(dispatch, getState, command) {
+    let currentChannel = getCurrentChannel(getState());
+    const currentTeamId = getCurrentTeamId(getState());
+
+    // Default to town square if there is no current channel (i.e., if Mattermost has not yet loaded)
+    if (!currentChannel) {
+        currentChannel = await Client4.getChannelByName(currentTeamId, 'town-square');
```
Comment (maisnamrajusingh) [inline webapp/src/actions/index.js:99]: Might be best to move the `town-square` into a variable. I remember this pattern being followed in some of the other plugins. @mickmister  what do you suggest ?

File: server/command.go (hunk 1 above)
Comment (sibasankarnayak) [inline server/command.go:256]: return responsef("We don't support re-queuing future items, only available for existing items.")

**Review by sibasankarnayak — APPROVED** [review]

File: webapp/src/index.js [hunk 3]
```diff
@@ -19,6 +19,10 @@ export default class Plugin {
             (channelId) => {
                 store.dispatch(openMeetingSettingsModal(channelId));
             });
+
+        registry.registerPostDropdownMenuAction('Re-queue', (params) => {
+            store.dispatch(requeueItem(params));
+        });
```
Comment (mickmister) [inline webapp/src/index.js:25]: We should only show this if the post has an agenda item right? We can use the third argument to [registerPostDropdownMenuAction](https://developers.mattermost.com/integrate/plugins/webapp/reference/#registerPostDropdownMenuAction) here.

Is it feasible to read the post's message and determine that way? We may not have the channel's agenda settings readily available. We may need to start putting something in the post's props to signal that this post is an agenda item if this isn't possible.

File: webapp/src/actions/index.js (hunk 2 above)
Comment (mickmister) [reply webapp/src/actions/index.js:99]: If we are using a post dropdown item to requeue this agenda item, Mattermost has indeed loaded and we can assume `currentChannel` is already a valid channel. We don't need this if block

File: webapp/src/actions/index.js [hunk 4]
```diff
@@ -80,4 +80,34 @@ export function performSearch(terms) {
 
         return dispatch(searchPostsWithParams(teamId, {terms, is_or_search: false, include_deleted_channels: viewArchivedChannels, page: 0, per_page: 20}, true));
     };
-}
\ No newline at end of file
+}
+
+export function requeueItem(itemId) {
+    return async (dispatch, getState) => {
+        const command = `/agenda requeue ${itemId}`;
```
Comment (mickmister) [inline webapp/src/actions/index.js:87]: What is `itemId`? It seems `requeueItem` is called like this:

```js
registry.registerPostDropdownMenuAction('Re-queue', (params) => {
    store.dispatch(requeueItem(params));
});
```

which would mean `params` is an `itemId`?

Comment (mickmister) [reply webapp/src/actions/index.js:87]: Will the user ever run `/agenda requeue` manually? If not, we should use an http endpoint for this feature and create a slash command. If the user would use this command manually, what would they supply for the `itemId`?

File: server/meeting_test.go [hunk 5]
```diff
@@ -138,7 +138,7 @@ func TestPlugin_GenerateHashtag(t *testing.T) {
 			jsonMeeting, err := json.Marshal(tt.args.meeting)
 			tAssert.Nil(err)
 			api.On("KVGet", tt.args.meeting.ChannelID).Return(jsonMeeting, nil)
-			got, err := mPlugin.GenerateHashtag(tt.args.meeting.ChannelID, tt.args.nextWeek, -1)
+			got, err := mPlugin.GenerateHashtag(tt.args.meeting.ChannelID, tt.args.nextWeek, -1, false, time.Now().Weekday()) // last parameter doesn't matter unless requeue
```
Comment (mickmister) [inline server/meeting_test.go:141]: Can we change the last parameter to not rely on `time.Now()`. Tests should have completely static data if possible. The comment here can be removed too.

File: server/meeting_test.go [hunk 6]
```diff
@@ -81,7 +81,7 @@ func TestPlugin_GenerateHashtag(t *testing.T) {
 					Schedule:      []time.Weekday{time.Monday},
 					HashtagFormat: "{{January 02 2006}}.vue",
 				}},
-			want:    "#" + assertNextWeekdayDate(time.Monday, false).Format("January 02 2006") + ".vue",
+			want:    "#" + strings.ReplaceAll(assertNextWeekdayDate(time.Monday, false).Format("January 02 2006"), " ", "_") + ".vue",
```
Comment (mickmister) [inline server/meeting_test.go:84]: We can use the underscores in the original formatting to avoid the string replace

```suggestion
			want:    "#" + assertNextWeekdayDate(time.Monday, false).Format("January_02_2006") + ".vue",
```

File: server/meeting_test.go [hunk 7]
```diff
@@ -67,9 +67,9 @@ func TestPlugin_GenerateHashtag(t *testing.T) {
 				meeting: &Meeting{
 					ChannelID:     "QA Backend",
 					Schedule:      []time.Weekday{time.Monday},
-					HashtagFormat: "QA-{{January 02 2006}}",
+					HashtagFormat: "QA_{{January 02 2006}}",
 				}},
-			want:    "#QA-" + assertNextWeekdayDate(time.Monday, true).Format("January 02 2006"),
+			want:    "#QA_" + strings.ReplaceAll(assertNextWeekdayDate(time.Monday, true).Format("January 02 2006"), " ", "_"),
```
Comment (mickmister) [inline server/meeting_test.go:72]: Why does `#QA-` need to be changed to `#QA_` here?

File: server/meeting_test.go [hunk 8]
```diff
@@ -176,7 +176,7 @@ func TestPlugin_GetMeeting(t *testing.T) {
 			},
 			want: &Meeting{
 				Schedule:      []time.Weekday{time.Thursday},
-				HashtagFormat: "Short-{{ Jan02 }}",
+				HashtagFormat: "Short_{{ Jan 2 }}",
```
Comment (mickmister) [inline server/meeting_test.go:179]: Same here, why the underscore?

File: server/meeting.go [hunk 9]
```diff
@@ -92,8 +103,10 @@ func (p *Plugin) GenerateHashtag(channelID string, nextWeek bool, weekday int) (
 		prefix = matchGroups[1]
 		hashtagFormat = strings.TrimSpace(matchGroups[2])
 		postfix = matchGroups[3]
+		formattedDate := meetingDate.Format(hashtagFormat)
+		formattedDate = strings.ReplaceAll(formattedDate, " ", "_")
```
Comment (mickmister) [inline server/meeting.go:107]: Why is this necessary? What about other formats that use a delimiter other than spaces and underscores?

File: server/meeting.go [hunk 10]
```diff
@@ -16,7 +16,7 @@ var (
 type Meeting struct {
 	ChannelID     string         `json:"channelId"`
 	Schedule      []time.Weekday `json:"schedule"`
-	HashtagFormat string         `json:"hashtagFormat"` // Default: {ChannelName}-Jan02
+	HashtagFormat string         `json:"hashtagFormat"` // Default: {ChannelName}-Jan-2
```
Comment (mickmister) [inline server/meeting.go:19]: Why did the default change?

File: server/meeting.go [hunk 11]
```diff
@@ -37,9 +37,10 @@ func (p *Plugin) GetMeeting(channelID string) (*Meeting, error) {
 		if err != nil {
 			return nil, err
 		}
+		paddedChannelName := strings.ReplaceAll(channel.Name, "-", "_")
 		meeting = &Meeting{
 			Schedule:      []time.Weekday{time.Thursday},
-			HashtagFormat: strings.Join([]string{fmt.Sprintf("%.15s", channel.Name), "{{ Jan02 }}"}, "-"),
+			HashtagFormat: strings.Join([]string{fmt.Sprintf("%.15s", paddedChannelName), "{{ Jan 2 }}"}, "_"),
```
Comment (mickmister) [inline server/meeting.go:43]: Why did the dash change to an underscore, and why did `Jan02` cange to `Jan 2`?

File: README.md [hunk 12]
```diff
@@ -27,10 +27,9 @@ The meeting settings for each channel can be configured in the Channel Header Dr
 
 Meeting settings include:
 
-- Schedule Day: Day of the week when the meeting is scheduled.
-- Hashtag Format: The format of the hashtag for the meeting date. The date format is based on [Go date and time formatting](https://yourbasic.org/golang/format-parse-string-time-date-example/#standard-time-and-date-formats).
-  The date format must be wrapped in double Braces ( {{ }} ).
-  A default is generated from the first 15 characters of the channel's name with the short name of the month and day (i.e. Dev-{{ Jan02 }}).
+- Schedule Day: Day of the week when the meeting is being scheduled.
+- Hashtag Format: The format of the hashtag for the meeting date. 
+  A default hashtag is generated from the first 15 characters of the channel's name with the short name of the month and day (i.e. Dev_{{ Jan 2 }}).
```
Comment (mickmister) [inline README.md:32]: Seems unrelated to this PR

File: server/meeting.go [hunk 13]
```diff
@@ -75,10 +76,20 @@ func (p *Plugin) GenerateHashtag(channelID string, nextWeek bool, weekday int) (
 			return "", err
 		}
 	} else {
-		// Get date for the list of days of the week
-		if meetingDate, err = nextWeekdayDateInWeek(meeting.Schedule, nextWeek); err != nil {
-			return "", err
+		// user didn't provide any specific date, Get date for the list of days of the week
```
Comment (mickmister) [inline server/meeting.go:79]:
```suggestion
		// User didn't provide any specific date, so we get the date for the list of days of the week
```

File: server/meeting.go [hunk 14]
```diff
@@ -75,10 +76,20 @@ func (p *Plugin) GenerateHashtag(channelID string, nextWeek bool, weekday int) (
...
+			}
+		} else {
+			if len(meeting.Schedule) == 1 && meeting.Schedule[0] == assignedDay { // if this day is the only day selected in settings
+				nextWeek = true
+			}
+			if meetingDate, err = nextWeekdayDateInWeekSkippingDay(meeting.Schedule, nextWeek, assignedDay); err != nil {
+				return "", err
+			}
 		}
+		//---- requeue Logic
```
Comment (mickmister) [inline server/meeting.go:92]: What does this comment mean?

File: server/command.go [hunk 15]
```diff
@@ -147,31 +164,147 @@ func (p *Plugin) executeCommandQueue(args *model.CommandArgs) *model.CommandResp
...
+	})
 	if appErr != nil {
-		return responsef("Error calculating list number")
+		return responsef("Error creating post: %s", appErr.Message)
+	}
+
+	return &model.CommandResponse{}
+}
+
+func calculateQueItemNumberAndUpdateOldItems(meeting *Meeting, args *model.CommandArgs, p *Plugin, hashtag string) (int, *model.CommandResponse) {
```
Comment (mickmister) [inline server/command.go:190]: Do we need to update old items in this PR?

File: server/command.go [hunk 16]
```diff
@@ -147,31 +164,147 @@ func (p *Plugin) executeCommandQueue(args *model.CommandArgs) *model.CommandResp
...
+	if errors != nil {
+		return errors
+	}
+
+	originalPostDate := strings.ReplaceAll(strings.TrimSpace(parsedMeetingMessage.date), "_", " ") // reverse what we do to make it a valid hashtag
+	originalPostMessage := strings.TrimSpace(parsedMeetingMessage.textMessage)
+
+	today := time.Now()
+	local, _ := time.LoadLocation("Local")
+	formattedDate, _ := time.ParseInLocation(hashtagDateFormat, originalPostDate, local)
```
Comment (mickmister) [inline server/command.go:249]: Do we use `Local` in other parts of this plugin?

File: server/command.go [hunk 17]
```diff
@@ -147,31 +164,147 @@ func (p *Plugin) executeCommandQueue(args *model.CommandArgs) *model.CommandResp
...
+		thisYear := today.Year()
+		formattedDate = formattedDate.AddDate(thisYear, 0, 0)
+	}
 
-	_, appErr = p.API.CreatePost(&model.Post{
+	if today.Year() <= formattedDate.Year() && today.YearDay() < formattedDate.YearDay() {
+		return responsef("We don't support re-queuing future items, only available for present and past items.")
+	}
+
+	hashtag, error := p.GenerateHashtag(args.ChannelId, false, -1, true, formattedDate.Weekday())
```
Comment (mickmister) [inline server/command.go:259]:
```suggestion
	hashtag, err := p.GenerateHashtag(args.ChannelId, false, -1, true, formattedDate.Weekday())
```

File: server/command.go [hunk 18]
```diff
@@ -147,31 +164,147 @@ func (p *Plugin) executeCommandQueue(args *model.CommandArgs) *model.CommandResp
...
 		ChannelId: args.ChannelId,
 		RootId:    args.RootId,
-		Message:   fmt.Sprintf("#### %v %v) %v", hashtag, numQueueItems+1, message),
+		Message:   fmt.Sprintf("#### %v %v) %v", hashtag, numQueueItems, originalPostMessage),
 	})
 	if appErr != nil {
-		return responsef("Error creating post: %s", appErr.Message)
+		return responsef("Error updating post: %s", appErr.Message)
 	}
+	return &model.CommandResponse{Text: fmt.Sprintf("Item has been Re-queued to %v", hashtag), ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL}
```
Comment (mickmister) [inline server/command.go:279]: We should use `responsef` here instead of `&model.CommandResponse`

File: server/command.go [hunk 19]
```diff
@@ -147,31 +164,147 @@ func (p *Plugin) executeCommandQueue(args *model.CommandArgs) *model.CommandResp
...
+func parseMeetingPost(meeting *Meeting, post *model.Post) (string, ParsedMeetingMessage, *model.CommandResponse) {
+	var (
+		prefix            string
+		hashtagDateFormat string
+	)
+	if matchGroups := meetingDateFormatRegex.FindStringSubmatch(meeting.HashtagFormat); len(matchGroups) == 4 {
+		prefix = matchGroups[1]
+		hashtagDateFormat = strings.TrimSpace(matchGroups[2])
+	} else {
+		return "", ParsedMeetingMessage{}, responsef("Error 267")
```
Comment (mickmister) [inline server/command.go:291]: What is `Error 267`?

File: server/command.go [hunk 20]
```diff
@@ -147,31 +164,147 @@ func (p *Plugin) executeCommandQueue(args *model.CommandArgs) *model.CommandResp
...
+	}
+
+	var (
+		messageRegexFormat = regexp.MustCompile(fmt.Sprintf(`(?m)^#### #%s(?P<date>.*) ([0-9]+)\) (?P<message>.*)?$`, prefix))
```
Comment (mickmister) [inline server/command.go:295]: Can this string be defined as a constant at the top of the file?

File: server/command.go [hunk 21]
```diff
@@ -147,31 +164,147 @@ func (p *Plugin) executeCommandQueue(args *model.CommandArgs) *model.CommandResp
...
+	matchGroups := messageRegexFormat.FindStringSubmatch(post.Message)
+	if len(matchGroups) == 4 {
+		parsedMeetingMessage := ParsedMeetingMessage{
+			date:        matchGroups[1],
+			number:      matchGroups[2],
+			textMessage: matchGroups[3],
+		}
+		return hashtagDateFormat, parsedMeetingMessage, nil
+	}
+	return hashtagDateFormat, ParsedMeetingMessage{}, responsef("Please ensure correct message format!")
```
Comment (mickmister) [inline server/command.go:307]: Under what circumstances will the user be receiving this message?

```suggestion
	return hashtagDateFormat, ParsedMeetingMessage{}, responsef("An error occurred processing the meeting hashtag.")
```

**Review by mickmister — CHANGES_REQUESTED** [review]
Great work @sanjaydemansol :+1:

I have some requests regarding the implementation, and removing code related to other PRs

**Comment by mickmister** [comment]
I accidentally reviewed the code for requeueing items because that is the majority of the PR. Please remove any code not related to changing the existing meeting items.

File: server/command.go (hunk 1 above)
Comment (sanjaydemansol) [reply server/command.go:256]: Sure, thanks.

File: webapp/src/actions/index.js (hunk 2 above)
Comment (sanjaydemansol) [reply webapp/src/actions/index.js:99]: All right.

File: webapp/src/components/meeting_settings/meeting_settings.jsx [hunk 22]
```diff
@@ -118,19 +129,50 @@ export default class MeetingSettingsModal extends React.PureComponent {
...
+                                    style={{height: '35px', border: '1px solid #ced4da'}}
+                                    className='form-select'
+                                >
+                                    <option value='Jan 2'>{'Month_day'}</option>
+                                    <option value='2 Jan'>{'day_Month'}</option>
+                                    <option value='1 2'>{'month_day'}</option>
+                                    <option value='2 1'>{'day_month'}</option>
+                                    <option value='2006 1 2'>{'year_month_day'}</option>
+
+                                </select>
```
[the remark this answers is not in the document]
Comment (sanjaydemansol) [reply webapp/src/components/meeting_settings/meeting_settings.jsx:163]: Please follow sequence for reviewing these three PRs  #66->#12->#47. There is one commit for each issue.

File: webapp/src/index.js (hunk 3 above)
Comment (sanjaydemansol) [reply webapp/src/index.js:25]: we could make channel's agenda settings readily available but 2/5 in my opinion we should go with your other suggestion i.e props in message. It would solve multiple issues like in current implementation we can't locate agenda messages if user change hashtag format multiple times. Also, backwards compatibility could be added after this.
I am not sure if that's possible, would be great if we could store something like this in props:
`{
isAgendaItem:true,
hastagDateFormat: "dd/mm/yy",
hashtagPrefix:"myprefix",
eventDate:"isoDate"
}`
@mickmister thoughts?

File: webapp/src/actions/index.js (hunk 4 above)
Comment (sanjaydemansol) [reply webapp/src/actions/index.js:87]: Sure will refactor `params` to `itemId`.
Made `/agenda requeue` to support future options like
`/agenda requeue all` # meeting canceled, I want to requeue all items to next meeting.
`/agenda requeue last` # requeue last items
`/agenda requeue last n` # requeue last n items

File: server/meeting_test.go (hunk 5 above)
Comment (sanjaydemansol) [reply server/meeting_test.go:141]: Not sure about the best way to do same, since go lang doesn't support default values for parameters.
any ideas?

File: server/meeting_test.go (hunk 6 above)
Comment (sanjaydemansol) [reply server/meeting_test.go:84]: Reason: `Format()` has special meaning for `_`.

File: server/meeting_test.go (hunk 7 above)
Comment (sanjaydemansol) [reply server/meeting_test.go:72]: Reason:
**Our plugin should be database agnostic.**


Screenshot_87
Ref. https://docs.mattermost.com/messaging/searching-in-mattermost.html

File: server/meeting_test.go (hunk 8 above)
Comment (sanjaydemansol) [reply server/meeting_test.go:179]: Reason in previous comment.

File: server/meeting.go (hunk 9 above)
Comment (sanjaydemansol) [reply server/meeting.go:107]: Considering `-` breaks on certain DB's,  `_` is the only safe character. But since `Format` which our plugin heavily rely on has special meaning `_`. I treated string before and after `Format`.

File: server/meeting.go (hunk 10 above)
Comment (sanjaydemansol) [reply server/meeting.go:19]: Same reason.

File: server/meeting.go (hunk 11 above)
Comment (sanjaydemansol) [reply server/meeting.go:43]: Explained.

File: README.md (hunk 12 above)
Comment (sanjaydemansol) [reply README.md:32]: Please follow sequence for reviewing these three PRs  #66->#12->#47. There is one commit for each issue.

File: server/meeting.go (hunk 13 above)
Comment (sanjaydemansol) [reply server/meeting.go:79]: Sure

File: server/meeting.go (hunk 14 above)
Comment (sanjaydemansol) [reply server/meeting.go:92]: sure, will remove it.

File: server/command.go (hunk 15 above)
Comment (sanjaydemansol) [reply server/command.go:190]: Please follow sequence for reviewing these three PRs  #66->#12->#47. There is one commit for each issue.

File: server/command.go (hunk 16 above)
Comment (sanjaydemansol) [reply server/command.go:249]: No, should I omit it?

File: server/command.go (hunk 17 above)
Comment (sanjaydemansol) [reply server/command.go:259]: Okay.

File: server/command.go (hunk 18 above)
Comment (sanjaydemansol) [reply server/command.go:279]: sure, will update.

File: server/command.go (hunk 20 above)
Comment (sanjaydemansol) [reply server/command.go:295]: sure.

File: server/command.go (hunk 19 above)
Comment (sanjaydemansol) [reply server/command.go:291]: Will make it friendly @maisnamrajusingh @mickmister .

File: server/command.go (hunk 21 above)
Comment (sanjaydemansol) [reply server/command.go:307]: If user tries to requeue non agenda message or message he/she has create with incorrect hashtag format.(we must remember user can always create agenda items manually given correct hashtag needs to be there)

**Review by hanzei — CHANGES_REQUESTED** [review]
Blocking until the re-queue changes have been removed

**Comment by sanjaydemansol** [comment]
> Blocking until the re-queue changes have been removed

Please follow sequence for reviewing these three PRs  #66->#12->#47. There is one commit for each issue.

File: webapp/src/components/meeting_settings/meeting_settings.jsx (hunk 22 above)
[the remark this answers is not in the document]
Comment (jfrerich) [reply webapp/src/components/meeting_settings/meeting_settings.jsx:163]: @sanjaydemansol, I'd prefer them as separate PR's in the future so we can view the code in isolation to the requested issue.

**Comment by jfrerich** [comment]
@sanjaydemansol, can you pull the code into 3 separate PRs, one for each issue?  #66, #12, and #47.

**Comment by jfrerich** [comment]
Actually, I see the PRs have been separated per issue.  The remaining work for this PR is to remove the code relative to PR #86 and #87

**Comment by sanjaydemansol** [comment]
> Actually, I see the PRs have been separated per issue. The remaining work for this PR is to remove the code relative to PR #86 and #87

Actually #86 is the base for #87 and #87 is base for this PR(#88).

**Comment by dipak-demansol** [comment]
@hanzei hii, you added the invalid label but i don't understand the label is for Dev or Qa, should i need to test this PR?

**Comment by hanzei** [comment]
@dipak-demansol No, there is nothing to test here

File: webapp/src/index.js (hunk 3 above)
Comment (mickmister) [reply webapp/src/index.js:25]: @sanjaydemansol Yeah in the props we can do something like:

```json
{
    "agendaSettings": {
        "hasftagDateFormat": "dd/mm/yy",
        "hashtagPrefix": "myprefix",
        "eventDate": "isoDate"
    }
}
```

Then we just need to check `post.props.agendaSettings`. If that's not in the props, don't show the requeue button.

What is the `eventDate` you're thinking of? Is that something we'll need to keep up to date in the post's props? When will we reference the props instead of the channel's agenda settings and vise versa?

File: webapp/src/actions/index.js (hunk 4 above)
Comment (mickmister) [reply webapp/src/actions/index.js:87]: Okay if this is the case, then maybe we should have a `/agenda requeue item (itemId)` command, and use that for the current programmatic case. Also what is `itemId`? Is it the post ID? Let's use `postId` as the identifier instead of `itemId` if this is the case.

File: server/meeting_test.go (hunk 5 above)
Comment (mickmister) [reply server/meeting_test.go:141]: Just use some random hardcoded day as the test data, and assert expected output based on that static input

File: server/meeting_test.go (hunk 7 above)
Comment (mickmister) [reply server/meeting_test.go:72]: @hanzei It seems we should be avoiding dashes and dots in all hashtags throughout plugins

File: server/meeting_test.go (hunk 6 above)
Comment (mickmister) [reply server/meeting_test.go:84]: How are existing agenda settings affected by this change?

File: server/meeting.go (hunk 11 above)
Comment (mickmister) [reply server/meeting.go:43]: Not sure why `Jan02` changed to `Jan 2` though

File: webapp/src/index.js (hunk 3 above)
Comment (sanjaydemansol) [reply webapp/src/index.js:25]: It will be the date the even is supposed to happen on, since hashtag format may be changed any time. Also user may change format after scheduling a meeting rendering old items unparsable. Ex, #1_2_2021 could mean any of two dates. So, either we must have old hashtag format(which was used at the time of creation of this hashtag) or better `eventDate`.

File: webapp/src/actions/index.js (hunk 4 above)
Comment (sanjaydemansol) [reply webapp/src/actions/index.js:87]: Right, will change it to `postId` and change command as well.

File: server/meeting_test.go (hunk 5 above)
Comment (sanjaydemansol) [reply server/meeting_test.go:141]: yes, did same in new PR.

File: server/meeting.go (hunk 11 above)
Comment (sanjaydemansol) [reply server/meeting.go:43]: Fixed it since most of our docs use Jan 2.
